### DISCUSSION TITLE
Outlook personnal account ?

### DISCUSSION THREAD

**Discussion by Norsagir** [simonrob/email-oauth2-proxy#296] [Q&A] [answered]
Hi !
I'm trying to configure this proxy as Vaultwarden does not want to implement OAuth2...
Microsoft has deactivated basic authentication for Microsoft Outlook personnal accounts.

So I'm trying to configure the proxy to use it with Vaultwarden.

I'm launching it via docker, with no-gui option and local server auth mode.
In the logs I see the URL I have to visit to authorize my account, but then Microsoft tells me :
`unauthorized_client: The client does not exist or is not enabled for consumers. If you are the application developer, configure a new application through the App Registrations in the Azure Portal at https://go.microsoft.com/fwlink/?linkid=2083908.`

Did I do something wrong? Or is this proxy useful for Office365 only and not outlook.com personnal accounts ?
Thanks for your help.

**Comment by simonrob** [comment, marked as answer]
Personal (i.e., free) accounts work, but you'll need to use an existing OAuth 2.0 client ID that is already admin approved by Microsoft as you can't add any yourself. See the proxy readme for links to existing open source clients you could adopt.

**Reply by Norsagir** [reply]
Thanks for your help.
I created another email address thanks to a small provider who has not banned basic authentication to solve my problem, because using a wrong client ID to configure your solution in order to configure Vautwarden creates a lot of dependencies. I think Vaultwarden devs have to implement Oauth2 authentication.
Thanks again !

**Reply by simonrob** [reply]
Just to follow up, it turns out that it _is_ possible to create your own OAuth 2.0 client for free/personal Outlook accounts. See #301 for details and a guide.
